keepalived/vrrp/vrrp_print.c

```diff
@@ -418,3 +421,182 @@ vrrp_print_stats(void)
 	}
 	fclose(file);
 }
+
+#ifdef _JSON_
+void
+vrrp_print_json(void)
+{
+	FILE *file;
+	file = fopen ("/tmp/keepalived.json","w");
+	if (!file) {
+		log_message(LOG_INFO, "Can't open /tmp/keepalived.json (%d: %s)",
+			errno, strerror(errno));
+		return;
+	}
+
+	list l = vrrp_data->vrrp;
+	if LIST_ISEMPTY(vrrp_data->vrrp) {
+		log_message(LOG_INFO, "vrrp_data->vrrp is empty");
+		return;
+	}
+
+	element e, f;
+	vrrp_t *vrrp;
+	struct json_object *array, *vips;
+	array = json_object_new_array();
+
+	for (e = LIST_HEAD(l); e; ELEMENT_NEXT(e)) {
+		vrrp = ELEMENT_DATA(e);
+		list v = vrrp->vip;
+		struct json_object *instance_json, *json_stats, *json_data;
+		instance_json = json_object_new_object();
+		json_stats = json_object_new_object();
+		json_data = json_object_new_object();
+		vips = json_object_new_array();
+
+		// Dump data to json
+		json_object_object_add(json_data, "iname",
+			json_object_new_string(vrrp->iname));
+		json_object_object_add(json_data, "dont_track_primary",
+			json_object_new_int(vrrp->dont_track_primary));
+		json_object_object_add(json_data, "skip_check_adv_addr",
+			json_object_new_int(vrrp->skip_check_adv_addr));
+		json_object_object_add(json_data, "strict_mode",
+			json_object_new_int(vrrp->strict_mode));
+		json_object_object_add(json_data, "vmac_flags",
+			json_object_new_int(vrrp->vmac_flags));
+		json_object_object_add(json_data, "vmac_ifname",
+			json_object_new_string(vrrp->vmac_ifname));
+		json_object_object_add(json_data, "master_priority",
+			json_object_new_int(vrrp->master_priority));
+		json_object_object_add(json_data, "last_transition",
+			json_object_new_int(vrrp->last_transition.tv_sec));
+		json_object_object_add(json_data, "garp_delay",
+			json_object_new_int(vrrp->garp_delay));
+		json_object_object_add(json_data, "garp_refresh",
+			json_object_new_int(vrrp->garp_refresh.tv_sec));
```

Comment:
It's not worth using timeval_to_double() here since it can only be a whole number of seconds, so just cast this to int to avoid the conversion warning.